perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/GroupsManagerMethod.java

```diff
@@ -308,27 +310,100 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
+	 *
+	 * @param group int Group <code>id</code>
+	 * @param members List<Integer> Members - can be empty <code>id</code>
+	 */
+	removeMembers {
+		@Override
+		public Object call(ApiCaller ac, Deserializer parms) throws PerunException {
+			ac.stateChangingCheck();
+			List<Integer> memberInts = parms.readList("members", Integer.class);
+			if (memberInts == null) {
+				throw new RpcException(RpcException.Type.MISSING_VALUE, "parameter 'members' not sent");
+			}
+			Group group = ac.getGroupById(parms.readInt("group"));
+			List<Member> members = new ArrayList<>();
+			for (Integer memberInt : memberInts) {
+				members.add(ac.getGroupsManager().getGroupMemberById(ac.getSession(), group, memberInt));
+			}
+			members = members.stream()
+				.filter(x -> MembershipType.DIRECT.equals(x.getMembershipType()))
+				.collect(Collectors.toList());
+			ac.getGroupsManager().removeMembers(ac.getSession(),
+				group,
+				members);
+			return null;
+		}
+	},
+
+
 	/*#
 	 * Removes a member from a group.
 	 *
+	 * @throw MemberNotExistsException When member doesn't exist
+	 * @throw NotGroupMemberException  When member is not in the group
+	 * @throw GroupNotExistsException When the group doesn't exist
+	 * @throw WrongAttributeAssignmentException When assigning atribute to wrong entity
+	 * @throw AttributeNotExistsException When attribute doesn't exist
+	 * @throw ExternallyManagedException When the group is externally managed
+	 *
 	 * @param group int Group <code>id</code>
 	 * @param member int Member <code>id</code>
 	 */
+	/*#
+	 * Removes a member from groups.
+	 *
+	 * @throw MemberNotExistsException When member doesn't exist
+	 * @throw NotGroupMemberException  When member is not in the group
+	 * @throw GroupNotExistsException When the group doesn't exist
+	 * @throw WrongAttributeAssignmentException When assigning atribute to wrong entity
+	 * @throw AttributeNotExistsException When attribute doesn't exist
+	 * @throw ExternallyManagedException When the group is externally managed
+	 *
+	 * @param groups List<Integer> Group - can be empty <code>id</code>
+	 * @param member int Member <code>id</code>
+	 */
 	removeMember {
-
 		@Override
 		public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
 			ac.stateChangingCheck();
-
-			ac.getGroupsManager().removeMember(ac.getSession(),
+			Member member = ac.getMemberById(parms.readInt("member"));
+			if (!MembershipType.DIRECT.equals(member.getMembershipType())) {
+				return null;
+			}
+			if (parms.contains("group")) {
+				ac.getGroupsManager().removeMember(ac.getSession(),
 					ac.getGroupById(parms.readInt("group")),
-					ac.getMemberById(parms.readInt("member")));
+					member);
+			} else if (parms.contains("groups")) {
+				List<Integer> groupsInts = parms.readList("groups", Integer.class);
+				List<Group> groups = new ArrayList<>();
+				for (Integer groupInt : groupsInts) {
+					groups.add(ac.getGroupById(groupInt));
```

Comment:
Here you should check, if member is not ``DIRECT`` in each group. If yes, then remove such group from the list. You can use ``ac.getGroupsManager().getGroupMemberById()`` as you use in ``removeMembers()``.